File: 3. DSA/02. C++/01. Basics/Question07.cpp
// Sum of first 'n' natural numbers 

#include <iostream>
using namespace std;

int main()
{
    int num;
    cout << "Enter a number: ";
    cin >> num;

    int sum = 0;
    for(int i = 1; i <= num; i++)
    {
        sum += i;
    }

    // int sum = num * (num + 1) / 2;

    cout << "\nThe sum of first " << num << " natural numbers is " << sum << endl;
}

File: 3. DSA/04. Python/01. Basics/Question07.py
# Sum of first 'n' natural numbers

n = int(input("Enter n: "))

#* Method 1
sum = n * (n + 1) // 2

#* Method 2
sum = 0
for i in range(n + 1):
    sum += i

print("Sum of first ", n, " natural numbers = ", sum, sep = '')